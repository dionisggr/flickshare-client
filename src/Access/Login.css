.login {
  display: flex;
  flex-direction: column;
  align-items: center;
  height: calc(100vh - 110px);
}

.login label {
  margin-top: 25px;
}

.login input {
  line-height: 2em;
  margin-top: 5px;
}

.login button {
  margin-top: 50px;
  padding: 10px 30px;
  border-radius: 5px;
  border: none;
  background-color: purple;
  color: white;
  font-size: 14px;
  font-weight: bold;
}