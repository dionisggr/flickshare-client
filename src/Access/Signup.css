.signup {
  display: flex;
  flex-direction: column;
  align-items: center;
  min-height: calc(100vh - 110px);
  margin-bottom: 100px;
}

.signup label {
  margin: 15px 0 3px 0;
  width: 70%;
  max-width: 300px;
  padding: 0 25px;
}

.signup input {
  line-height: 2.5em;
  width: 70%;
  max-width: 300px;
  border-radius: 20px;
  border: solid 2px rgb(61, 0, 61);
  padding-left: 15px;
  font-size: 15px;
  outline: none;
}

#error_username, #error_password, #error_repeat_password {
  color: red;
}

.signup button {
  background-color: rgba(2, 214, 2, 0.842);
  outline: none;
  border: solid 1px rgba(0, 0, 0, 0.712);
  border-radius: 5px;
  margin-top: 50px;
  padding: 5px 35px;
  font-weight: bold;
  color: white;
  min-width: fit-content;
  line-height: 1.8em;
}

.signup button:hover {
  background-color: rgb(25, 231, 25);
}

.signup button:active {
  transform: scale(0.95);
}

.error {
  color: red;
  padding: 0;
  margin: 0;
  height: fit-content;
  padding: 0 10%;
  font-size: 14px;
}