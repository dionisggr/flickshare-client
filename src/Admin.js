import React from 'react';
import { JWT_SECRET } from './config';
import jwt from 'jsonwebtoken';
import ListPreview from './ListPreview';
import Error from './Error';
import api from './api';
import './Admin.css';

class Admin extends React.Component {
  state = { lists: [] };

  componentDidMount() {
    const flickshareToken = JSON.parse(window.localStorage.getItem('flickshareToken'));

    const decoded = (flickshareToken)
      ? jwt.verify(flickshareToken, JWT_SECRET, (error, decoded) => {
          if (error) return null;
          return decoded;
        })
      : null;

    const admin = (decoded) ? decoded.admin : false;
    const user_id = (decoded) ? parseInt(decoded.user_id) : null;
      
    if (admin) {
      api.getAllLists()
        .then(lists => {
          this.setState({ lists });
        })
        .catch(error => <Error message={error} />)
    };
  };

  render() {
    const flickshareToken = JSON.parse(window.localStorage.getItem('flickshareToken'));

    const decoded = (flickshareToken)
      ? jwt.verify(flickshareToken, JWT_SECRET, (error, decoded) => {
          if (error) return null;
          return decoded;
        })
      : null;
    
    if (!decoded.admin) {
      return <Error message='Unauthorized access.' />;
    };
    
    const { lists } = this.state;
     
    return (
      <div className='admin-lists'>
        <h3>All Lists</h3>
        {
          lists.map(list =>  
            <ListPreview
              key={list.list_id}
              list={list}
            />
          )
        }
      </div>
    );
  };
};

export default Admin;