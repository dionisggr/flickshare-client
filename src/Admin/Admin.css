.admin-lists {
  height: calc(100% - 110px);
  margin-bottom: 100px;
}

.admin-lists h3 {
  text-align: center;
}