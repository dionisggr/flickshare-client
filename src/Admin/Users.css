.users {
  display: flex;
  flex-direction: column;
  align-items: center;
  min-height: calc(100vh - 235px);
  margin-bottom: 100px;
}

.home-button {
  background-color: rgba(2, 214, 2, 0.842);
  outline: none;
  border: solid 1px rgba(0, 0, 0, 0.712);
  border-radius: 5px;
  margin-top: 35px;
  padding: 5px 35px;
  font-weight: bold;
  color: white;
  min-width: fit-content;
  text-decoration: none;
}

.home-button:hover {
  background-color: rgb(25, 231, 25);
}

.home-button:active {
  transform: scale(0.95);
}