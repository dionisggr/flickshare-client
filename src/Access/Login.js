import React from 'react';
import api from '../services/api';
import './Login.css';

class Login extends React.Component {
  static defaultProps = { userLogged: {} };

  login = async (evt) => {
    evt.preventDefault();

    const username = evt.target.username.value;
    const password = evt.target.password.value;

    const { history, userLogged, setIdleTimer } = this.props;

    const response = await api.login(username, password)
      .catch(error => console.log({ error }));
    
    if (response) {
      const { flickshareToken } = response;

      window.localStorage.setItem('flickshareToken', JSON.stringify(flickshareToken));
  
      userLogged(true);
      setIdleTimer();
  
      history.push('/home');

    } else {
      evt.target.querySelector('#error_login').style.display = 'block';
      evt.target.querySelector('#username').value = '';
      evt.target.querySelector('#password').value = '';
    };
  };

  render() {
    return (
      <form
        id='login'
        className='login'
        onSubmit={this.login}>

        <h3>LOGIN</h3>

        <label htmlFor='username'>Username</label>

        <input
          type='text' required
          name='username' id='username'
        />

        <label htmlFor='password'>Password</label>

        <input
          type='text' required
          name='password' id='password'
        />
        <label
          htmlFor='login'
          id='error_login'
          className='error'
          style={{ display: 'none' }}
        >
          <p>Invalid credentials.</p>
        </label>

        <button type='submit'>Submit</button>

      </form>
    );
  };
};

export default Login;