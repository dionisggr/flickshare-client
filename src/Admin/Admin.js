import React from 'react';
import { JWT_SECRET } from '../config';
import jwt from 'jsonwebtoken';
import ListPreview from '../Lists/List/ListPreview';
import Error from '../error-handlers/Error';
import api from '../services/api';
import './Admin.css';

class Admin extends React.Component {
  state = { lists: [] };

  componentDidMount() {
    const flickshareToken = JSON.parse(window.localStorage.getItem('flickshareToken'));

    const decoded = (flickshareToken)
      ? jwt.verify(flickshareToken, JWT_SECRET, (error, decoded) => {
          if (error) return null;
          return decoded;
        })
      : null;

    const admin = (decoded) ? decoded.admin : false;
      
    if (admin) {
      api.getAllLists()
        .then(lists => {
          this.setState({ lists });
        })
        .catch(error => <Error message={error} />)
    };
  };

  render() {
    const flickshareToken = JSON.parse(window.localStorage.getItem('flickshareToken'));

    const decoded = (flickshareToken)
      ? jwt.verify(flickshareToken, JWT_SECRET, (error, decoded) => {
          if (error) return null;
          return decoded;
        })
      : null;
    
    if (decoded && !decoded.admin) {
      return <Error message='Unauthorized access.' />;
    };
    
    const { lists } = this.state;
     
    return (
      <div className='admin-lists'>
        <h3>All Lists</h3>
        {
          lists.map(list =>  
            <ListPreview
              key={list.list_id}
              list={list}
            />
          )
        }
      </div>
    );
  };
};

export default Admin;