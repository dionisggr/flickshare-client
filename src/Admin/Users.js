import React from 'react';
import { Link } from 'react-router-dom';
import Error from '../error-handlers/Error';
import api from '../services/api';
import UserPreview from '../Users/UserPreview';
import './Users.css';

class Users extends React.Component {
  state = { users: [] };

  componentDidMount() {
    api.getAllUsers()
      .then(users => {
        this.setState({ users });
      })
      .catch(error => <Error message={error} />);
  };

  render() {

    const { users } = this.state;
    return (
      <div className='users'>
        <h3>Users</h3>
        {
          users.map((user, idx) =>
            <UserPreview
              key={idx}
              user={user}
            />
          )
        }

        <Link
          to='/admin'
          className='home-button'
        >Home</Link>
      </div>
    );
  };
};

export default Users;