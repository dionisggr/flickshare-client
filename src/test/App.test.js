import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Header from '../Header/Header';
import Footer from '../Footer/Footer';
import Welcome from '../Main/Welcome';
import Signup from '../Access/Signup'; 
import Login from '../Access/Login'; 
import MainMenu from '../Main/MainMenu';
import Lists from '../Lists/Lists';
import List from '../Lists/List/List';
import ListPreview from '../Lists/List/ListPreview';
import ListEdit from '../Lists/List/ListEdit';
import Suggestions from '../Lists/Suggestions/Suggestions';
import MovieSearch from '../Movies/MovieSearch';
import Movie from '../Movies/Movie';
import MoviePreview from '../Movies/MoviePreview';
import MovieOptions from '../Movies/MovieOptions';
import Users from '../Admin/Users'; 
import User from '../Users/User'; 
import UserEdit from '../Users/UserEdit'; 
import Admin from '../Admin/Admin'
import ResetPassword from '../Users/ResetPassword'; 
import App from '../App';

describe('The App', () => {
  const app = <BrowserRouter> <App /> </BrowserRouter>;
  global.scroll = jest.fn();

  describe('Smoke Test', () => {
    it('renders learn react link', () => {
      render(app);
    });
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(app, div);
      ReactDOM.unmountComponentAtNode(div);
    });
  });
  describe('Snapshot Test', () => {
    it('renders the UI as expected', () => {
      const tree = renderer.create(app).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});

describe('The Components', () => {
  describe('The Header', () => {
    const header = <BrowserRouter> <Header /> </BrowserRouter>;

    it('renders without crashing', () => {
      const div = document.createElement('div');

      ReactDOM.render(header, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    it('renders the UI as expected', () => {
      const tree = renderer.create(header).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('The Footer', () => {
    it('renders without crashing', () => {
      const div = document.createElement('div');

      ReactDOM.render(<Footer />, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    it('renders the UI as expected', () => {
      const tree = renderer.create(<Footer />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('The Welcome Page', () => {
    const welcome = <BrowserRouter><Welcome /></BrowserRouter>;

    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(welcome, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    it('renders the UI as expected', () => {
      const tree = renderer.create(welcome).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('The Signup Page', () => {
    const signup = <BrowserRouter><Signup /></BrowserRouter>;

    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(signup, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    it('renders the UI as expected', () => {
      const tree = renderer.create(signup).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('The Login Page', () => {
    const login = <BrowserRouter><Login /></BrowserRouter>;

    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(login, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    it('renders the UI as expected', () => {
      const tree = renderer.create(login).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('The Main Menu', () => {
    const mainMenu = <BrowserRouter><MainMenu /></BrowserRouter>;

    it('renders without crashing', () => {

      const div = document.createElement('div');
      ReactDOM.render(mainMenu, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    it('renders the UI as expected', () => {
      const tree = renderer.create(mainMenu).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('User-Related', () => {
    describe('The Users', () => {
      const users = <BrowserRouter> <Users /> </BrowserRouter>;
      it('render without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(users, div);
        ReactDOM.unmountComponentAtNode(div);
      });
      it('render the UI as expected', () => {
        const tree = renderer.create(users).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
    describe('The User Page', () => {
      it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<User history={{}} />, div);
        ReactDOM.unmountComponentAtNode(div);
      });
      it('renders the UI as expected', () => {
        const tree = renderer.create(<User history={{}} />).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
    describe('The Edit User Page', () => {
      it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<UserEdit />, div);
        ReactDOM.unmountComponentAtNode(div);
      });
      it('renders the UI as expected', () => {
        const tree = renderer.create(<UserEdit />).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
    describe('The Reset Password Page', () => {
      it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ResetPassword />, div);
        ReactDOM.unmountComponentAtNode(div);
      });
      it('renders the UI as expected', () => {
        const tree = renderer.create(<ResetPassword />).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
  describe('List-Related', () => {
    describe('The Lists', () => {
      it('render without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Lists />, div);
        ReactDOM.unmountComponentAtNode(div);
      });
      it('render the UI as expected', () => {
        const tree = renderer.create(<Lists />).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
    describe('The List Page', () => {
    const list = <BrowserRouter><List /></BrowserRouter>;

      it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(list, div);
        ReactDOM.unmountComponentAtNode(div);
      });
      it('renders the UI as expected', () => {
        const tree = renderer.create(list).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
    describe('The List Preview', () => {
    const listPreview = <BrowserRouter><ListPreview /></BrowserRouter>;

      it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(listPreview, div);
        ReactDOM.unmountComponentAtNode(div);
      });
      it('renders the UI as expected', () => {
        const tree = renderer.create(listPreview).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
    describe('The Edit List Page', () => {
      const listEdit = <BrowserRouter><ListEdit /></BrowserRouter>;

      it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(listEdit, div);
        ReactDOM.unmountComponentAtNode(div);
      });
      it('renders the UI as expected', () => {
        const tree = renderer.create(listEdit).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
    describe('The Suggestions Page', () => {
      const suggestions = <BrowserRouter><Suggestions /></BrowserRouter>;

      it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(suggestions, div);
        ReactDOM.unmountComponentAtNode(div);
      });
      it('renders the UI as expected', () => {
        const tree = renderer.create(suggestions).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
  describe('Movie-Related', () => {
    describe('The Movie Search Page', () => {
      const movieSearch = <BrowserRouter> <MovieSearch /> </BrowserRouter>;

      it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(movieSearch, div);
        ReactDOM.unmountComponentAtNode(div);
      });
      it('renders the UI as expected', () => {
        const tree = renderer.create(movieSearch).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
    describe('The Movie Page', () => {
      const movie = <BrowserRouter> <Movie /> </BrowserRouter>;

      it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(movie, div);
        ReactDOM.unmountComponentAtNode(div);
      });
      it('renders the UI as expected', () => {
        const tree = renderer.create(movie).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
    describe('The Movie Preview', () => {
      it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<MoviePreview />, div);
        ReactDOM.unmountComponentAtNode(div);
      });
      it('renders the UI as expected', () => {
        const tree = renderer.create(<MoviePreview />).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
    describe('The Movie Options', () => {
      const movieOptions = <BrowserRouter> <MovieOptions movie={{}} /> </BrowserRouter>;

      it('render without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(movieOptions, div);
        ReactDOM.unmountComponentAtNode(div);
      });
      it('render the UI as expected', () => {
        const tree = renderer.create(movieOptions).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
  describe('Admin-Related', () => {
    describe('The Admin Page (All Lists)', () => {
      it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Admin />, div);
        ReactDOM.unmountComponentAtNode(div);
      });
      it('renders the UI as expected', () => {
        const tree = renderer.create(<Admin />).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
    describe('The Users Page (All Users)', () => {
      const users = <BrowserRouter> <Users /> </BrowserRouter>;
      it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(users, div);
        ReactDOM.unmountComponentAtNode(div);
      });
      it('renders the UI as expected', () => {
        const tree = renderer.create(users).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});


