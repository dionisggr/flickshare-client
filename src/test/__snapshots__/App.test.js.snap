// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The App Snapshot Test renders the UI as expected 1`] = `
Array [
  " ",
  <main
    className="App"
  >
    <header>
      <h1>
        <a
          href="/"
          onClick={[Function]}
        >
          FlickShare
        </a>
      </h1>
      <nav>
        <ul>
          <li>
            <a
              href="/register"
              onClick={[Function]}
            >
              Register
            </a>
          </li>
          <li>
            <a
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
        </ul>
      </nav>
    </header>
    <div
      className="welcome"
    >
      <div
        className="with-background"
      >
        <h3>
          Welcome to FlickShare!
        </h3>
        <label>
          Why struggle choosing a movie to watch?
        </label>
        <p>
          Good movie suggestions are hard to come by. Usually we rely on the Internet or our friends to decide what to watch next.
        </p>
        <p>
          What if we could combine both?
        </p>
        <p>
          What if we could create a list with our favorite movies and be suggested other movies based on your likes?
        </p>
        <p>
          What if we could take this step further, and base suggestions on your friends lists as well?
        </p>
        <label>
          It's simple:
        </label>
        <ol>
          <li>
            Create an account.
          </li>
          <li>
            Create a list with your favorite movies.
          </li>
          <li>
            Make new friends, watch other users' lists and send suggestions!
          </li>
          <li>
            Come back for more!
          </li>
        </ol>
        <a
          href="/register"
          onClick={[Function]}
        >
          Create a new account!
        </a>
        <a
          href="/home"
          onClick={[Function]}
        >
          I just want to browse.
        </a>
      </div>
    </div>
    <footer>
      <div
        className="icons"
      >
        <a
          aria-label="email"
          href="mailto:dionisggr@gmail.com"
          target="_blank"
        >
          <i
            aria-hidden="true"
            className="fa fa-envelope"
          />
        </a>
        <a
          aria-label="github"
          href="https://github.com/dionisggr/flickshare-client/"
          target="_blank"
        >
          <i
            aria-hidden="true"
            className="fa fa-github"
          />
        </a>
        <a
          aria-label="linked-in"
          href="https://www.linkedin.com/in/dionis-gonzalez-ramirez/"
          target="_blank"
        >
          <i
            aria-hidden="true"
            className="fa fa-linkedin-square"
          />
        </a>
      </div>
      <h2>
        © Dionis González Ramírez
      </h2>
    </footer>
  </main>,
  " ",
]
`;

exports[`The Components Admin-Related The Admin Page (All Lists) renders the UI as expected 1`] = `
<div
  className="admin-lists"
>
  <h3>
    All Lists
  </h3>
</div>
`;

exports[`The Components Admin-Related The Users Page (All Users) renders the UI as expected 1`] = `
Array [
  " ",
  <div
    className="users"
  >
    <h3>
      Users
    </h3>
    <a
      className="home-button"
      href="/admin"
      onClick={[Function]}
    >
      HOME
    </a>
  </div>,
  " ",
]
`;

exports[`The Components List-Related The Edit List Page renders the UI as expected 1`] = `
<form
  className="list-edit"
  onSubmit={[Function]}
>
  <label
    htmlFor="list_name"
  >
    Name:
  </label>
  <input
    id="list_name"
    name="list_name"
    onChange={[Function]}
    required={true}
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    Create
  </button>
</form>
`;

exports[`The Components List-Related The List Page renders the UI as expected 1`] = `
<div
  className="list"
>
  <h3>
    List Details
  </h3>
  <div
    className="list-movies"
  >
    <div
      className="list-preview"
    >
      <a
        href="/lists/undefined"
        onClick={[Function]}
      >
        <h3 />
      </a>
      <div
        className="list-preview-movies"
      >
        <div
          className="no-movies"
        >
          <h3>
            Wow, so much empty.
          </h3>
        </div>
      </div>
    </div>
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    BACK
  </button>
</div>
`;

exports[`The Components List-Related The List Preview renders the UI as expected 1`] = `
<div
  className="list-preview"
>
  <a
    href="/lists/undefined"
    onClick={[Function]}
  >
    <h3 />
  </a>
  <div
    className="list-preview-movies"
  >
    <div
      className="no-movies"
    >
      <h3>
        Wow, so much empty.
      </h3>
    </div>
  </div>
</div>
`;

exports[`The Components List-Related The Lists render the UI as expected 1`] = `
<div
  className="lists"
>
  <h3>
    My Lists
  </h3>
  <button
    onClick={[Function]}
    type="button"
  >
    Add
  </button>
</div>
`;

exports[`The Components List-Related The Suggestions Page renders the UI as expected 1`] = `
<div
  className="error"
>
  <h3>
    Error
  </h3>
  <h4>
    Unauthorized request.
  </h4>
  <a
    href="/home"
    onClick={[Function]}
  >
    HOME
  </a>
</div>
`;

exports[`The Components Movie-Related The Movie Options render the UI as expected 1`] = `
Array [
  " ",
  <div
    className="movie-preview-options"
  >
    <a
      href="/movies/undefined"
      onClick={[Function]}
    >
      More Info
    </a>
  </div>,
  " ",
]
`;

exports[`The Components Movie-Related The Movie Page renders the UI as expected 1`] = `
Array [
  " ",
  <div
    className="movie"
  >
    <img
      alt="undefined poster"
    />
    <div
      className="movie-data"
    >
      <label>
        <b>
          Name: 
        </b>
      </label>
      <label>
        <b>
          Description: 
        </b>
      </label>
      <label>
        <b>
          Release Date: 
        </b>
      </label>
      <label>
        <b>
          Average Vote: 
        </b>
      </label>
      <label>
        <b>
          Vote Count: 
        </b>
        NaN
      </label>
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      Add to List
    </button>
  </div>,
  " ",
]
`;

exports[`The Components Movie-Related The Movie Preview renders the UI as expected 1`] = `
<div
  className="movie-preview"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label />
  <img
    alt="undefined poster"
  />
</div>
`;

exports[`The Components Movie-Related The Movie Search Page renders the UI as expected 1`] = `
Array [
  " ",
  <div
    className="movie-search"
  >
    <form
      className="search"
      onSubmit={[Function]}
    >
      <h3>
        Movie Search
      </h3>
      <input
        id="movie_input"
        name="movie_input"
        placeholder="Search..."
        type="text"
      />
      <button
        type="submit"
      >
        Search
      </button>
    </form>
    <div
      className="results"
    />
  </div>,
  " ",
]
`;

exports[`The Components The Footer renders the UI as expected 1`] = `
<footer>
  <div
    className="icons"
  >
    <a
      aria-label="email"
      href="mailto:dionisggr@gmail.com"
      target="_blank"
    >
      <i
        aria-hidden="true"
        className="fa fa-envelope"
      />
    </a>
    <a
      aria-label="github"
      href="https://github.com/dionisggr/flickshare-client/"
      target="_blank"
    >
      <i
        aria-hidden="true"
        className="fa fa-github"
      />
    </a>
    <a
      aria-label="linked-in"
      href="https://www.linkedin.com/in/dionis-gonzalez-ramirez/"
      target="_blank"
    >
      <i
        aria-hidden="true"
        className="fa fa-linkedin-square"
      />
    </a>
  </div>
  <h2>
    © Dionis González Ramírez
  </h2>
</footer>
`;

exports[`The Components The Header renders the UI as expected 1`] = `
Array [
  " ",
  <header>
    <h1>
      <a
        href="/"
        onClick={[Function]}
      >
        FlickShare
      </a>
    </h1>
    <nav>
      <ul>
        <li>
          <a
            href="/register"
            onClick={[Function]}
          >
            Register
          </a>
        </li>
        <li>
          <a
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  " ",
]
`;

exports[`The Components The Login Page renders the UI as expected 1`] = `
<form
  className="login"
  id="login"
  onSubmit={[Function]}
>
  <h3>
    LOGIN
  </h3>
  <label
    htmlFor="username"
  >
    Username
  </label>
  <input
    autoComplete="off"
    id="username"
    name="username"
    required={true}
    type="text"
  />
  <label
    htmlFor="password"
  >
    Password
  </label>
  <input
    id="password"
    name="password"
    required={true}
    type="password"
  />
  <label
    className="error"
    htmlFor="login"
    id="error_login"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <p>
      Invalid credentials.
    </p>
  </label>
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`The Components The Main Menu renders the UI as expected 1`] = `
<div
  className="main-menu"
>
  <ul>
    <li>
      <a
        href="/users/null/suggestions"
        onClick={[Function]}
      >
        Suggestions
      </a>
    </li>
    <li>
      <a
        href="/users/null/lists"
        onClick={[Function]}
      >
        My Lists
      </a>
    </li>
    <li>
      <a
        href="/movie/search"
        onClick={[Function]}
      >
        Search
      </a>
    </li>
  </ul>
</div>
`;

exports[`The Components The Register Page renders the UI as expected 1`] = `
<form
  autoComplete="off"
  className="register"
  onSubmit={[Function]}
>
  <h3>
    REGISTER
  </h3>
  <label
    htmlFor="first_name"
  >
    First Name:
  </label>
  <input
    autoComplete="none"
    id="first_name"
    name="first_name"
    required={true}
    type="text"
  />
  <label
    htmlFor="last_name"
  >
    Last Name:
  </label>
  <input
    autoComplete="none"
    id="last_name"
    name="last_name"
    required={true}
    type="text"
  />
  <label
    htmlFor="email"
  >
    E-mail:
  </label>
  <input
    autoComplete="none"
    id="email"
    name="email"
    required={true}
    type="text"
  />
  <label
    htmlFor="username"
  >
    Username:
  </label>
  <input
    autoComplete="none"
    id="username"
    name="username"
    required={true}
    type="text"
  />
  <label
    className="error"
    htmlFor="username"
    id="error_username"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <p>
      Username exists.
    </p>
  </label>
  <label
    htmlFor="password"
  >
    Password
  </label>
  <input
    id="password"
    name="password"
    required={true}
    type="password"
  />
  <label
    className="error"
    htmlFor="password"
    id="error_password"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <p>
      Password must be at least 8 charaters.
    </p>
    <p>
      Password must not contain spaces.
    </p>
    <p>
      Password must contain at least 1 Uppercase.
    </p>
    <p>
      Password must contain at least 1 number.
    </p>
  </label>
  <label
    htmlFor="repeat_password"
  >
    Repeat password:
  </label>
  <input
    id="repeat_password"
    name="repeat_password"
    required={true}
    type="password"
  />
  <label
    className="error"
    htmlFor="password"
    id="error_repeat_password"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <p>
      Passwords must match.
    </p>
  </label>
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`The Components The Welcome Page renders the UI as expected 1`] = `
<div
  className="welcome"
>
  <div
    className="with-background"
  >
    <h3>
      Welcome to FlickShare!
    </h3>
    <label>
      Why struggle choosing a movie to watch?
    </label>
    <p>
      Good movie suggestions are hard to come by. Usually we rely on the Internet or our friends to decide what to watch next.
    </p>
    <p>
      What if we could combine both?
    </p>
    <p>
      What if we could create a list with our favorite movies and be suggested other movies based on your likes?
    </p>
    <p>
      What if we could take this step further, and base suggestions on your friends lists as well?
    </p>
    <label>
      It's simple:
    </label>
    <ol>
      <li>
        Create an account.
      </li>
      <li>
        Create a list with your favorite movies.
      </li>
      <li>
        Make new friends, watch other users' lists and send suggestions!
      </li>
      <li>
        Come back for more!
      </li>
    </ol>
    <a
      href="/register"
      onClick={[Function]}
    >
      Create a new account!
    </a>
    <a
      href="/home"
      onClick={[Function]}
    >
      I just want to browse.
    </a>
  </div>
</div>
`;

exports[`The Components User-Related The Edit User Page renders the UI as expected 1`] = `
<form
  autoComplete="off"
  className="user-edit"
  onSubmit={[Function]}
>
  <h3>
    MY ACCOUNT
  </h3>
  <img
    alt=""
    src=""
  />
  <label
    htmlFor="first_name"
  >
    First Name:
  </label>
  <input
    id="first_name"
    name="first_name"
    onChange={[Function]}
    type="text"
  />
  <label
    htmlFor="last_name"
  >
    Last Name:
  </label>
  <input
    id="last_name"
    name="last_name"
    onChange={[Function]}
    type="text"
  />
  <label
    htmlFor="email"
  >
    E-mail:
  </label>
  <input
    id="email"
    name="email"
    onChange={[Function]}
    type="text"
  />
  <label
    htmlFor="username"
  >
    Username:
  </label>
  <input
    id="username"
    name="username"
    onChange={[Function]}
    type="text"
  />
  <button
    type="submit"
  >
    Submit
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Back
  </button>
</form>
`;

exports[`The Components User-Related The Reset Password Page renders the UI as expected 1`] = `
<form
  className="reset-password"
  onSubmit={[Function]}
>
  <h3>
    RESET PASSWORD
  </h3>
  <label
    htmlFor="password"
  >
    New Password
  </label>
  <input
    id="password"
    name="password"
    required={true}
    type="password"
  />
  <label
    className="error"
    htmlFor="password"
    id="error_password"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <p>
      Password must be at least 8 charaters.
    </p>
    <p>
      Password must not contain spaces.
    </p>
    <p>
      Password must contain at least 1 Uppercase.
    </p>
    <p>
      Password must contain at least 1 number.
    </p>
  </label>
  <label
    htmlFor="password"
  >
    Repeat Password
  </label>
  <label
    className="error"
    htmlFor="password"
    id="error_repeat_password"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <p>
      Passwords must match.
    </p>
  </label>
  <input
    id="repeat_password"
    name="repeat_password"
    required={true}
    type="password"
  />
  <button
    type="submit"
  >
    Reset
  </button>
</form>
`;

exports[`The Components User-Related The User Page renders the UI as expected 1`] = `
<div
  className="user"
>
  <h3>
    User Account
  </h3>
  <img
    alt="undefined's avatar"
    src="https://robohash.org/undefined"
  />
  <div
    className="user-info"
  >
    <label>
      <b>
        Username: 
      </b>
    </label>
    <label>
      <b>
        Email: 
      </b>
    </label>
    <label>
      <b>
        First Name: 
      </b>
    </label>
    <label>
      <b>
        Last Name: 
      </b>
    </label>
  </div>
  <button
    className="lists-button"
    onClick={[Function]}
    type="button"
  >
    LISTS
  </button>
  <button
    className="reset-password-button"
    onClick={[Function]}
    type="button"
  >
    RESET PASSWORD
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    EDIT
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    DELETE ACCOUNT
  </button>
  <button
    type="button"
  >
    BACK
  </button>
</div>
`;

exports[`The Components User-Related The Users render the UI as expected 1`] = `
Array [
  " ",
  <div
    className="users"
  >
    <h3>
      Users
    </h3>
    <a
      className="home-button"
      href="/admin"
      onClick={[Function]}
    >
      HOME
    </a>
  </div>,
  " ",
]
`;
